core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -67,50 +66,45 @@ public static void check(final Settings settings) {
      * @param enforceLimits true if the checks should be enforced or
      *                      warned
      * @param checks        the checks to execute
+     * @param nodeName      the node name to be used as a logging prefix
      */
     // visible for testing
-    static void check(final boolean enforceLimits, final List<Check> checks) {
-        final ESLogger logger = Loggers.getLogger(BootstrapCheck.class);
+    static void check(final boolean enforceLimits, final List<Check> checks, final String nodeName) {
+        final ESLogger logger = Loggers.getLogger(BootstrapCheck.class, nodeName);
+
+        final List<IllegalStateException> exceptions = new ArrayList<>();
 
         for (final Check check : checks) {
             final boolean fail = check.check();
             if (fail) {
                 if (enforceLimits) {
-                    throw new RuntimeException(check.errorMessage());
+                    exceptions.add(new IllegalStateException(check.errorMessage()));
```

Comment:
Nit: I think it will be cleaner to collect messages and in the end either log them all or suppress them all and rethrow